runtime/cudaq/utils/cudaq_utils.h

```diff
@@ -240,12 +240,30 @@ std::vector<double> random_vector(const double l_range, const double r_range,
                                   const std::size_t size,
                                   const uint32_t seed = std::random_device{}());
 
-inline std::vector<std::size_t> range(int N) {
-  std::vector<std::size_t> vec(N);
-  std::iota(vec.begin(), vec.end(), 0);
+/// @brief Return a vector of integers. The first element is the
+/// user-specified `start` value. The remaining values are all values
+/// incremented by `step` (defaults to 1) until the `stop` value is reached
+/// (exclusive).
+template <typename ElementType>
+inline std::vector<ElementType> range(ElementType start, ElementType stop,
+                                      ElementType step = 1) {
```

Comment:
Did you update the bridge for this? When I implemented this in the bridge I remember wondering about the default step value.

Reply:
I did not. I thought I would leave that for a follow up. I'll add the ticket.